### DISCUSSION TITLE
"Devtools failed to load source map"

### DISCUSSION THREAD

**Discussion by richardsprague** [quarto-dev/quarto-cli#5793] [Q&A] [answered]
### Description

My Javascript console presents this error for every one of my Quarto-generated sites. 

```
DevTools failed to load source map: Could not load content for https://diycgm.com/site_libs/bootstrap/bootstrap.bundle.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load source map: Could not load content for https://diycgm.com/site_libs/bootstrap/603954f6f730b7a48ae583e90c07e56e.css.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load source map: Could not load content for https://diycgm.com/site_libs/quarto-html/debc5d5d77c3f9108843748ff7464032.css.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```
ChatGPT says
>  indicates that there is an issue with the source map file itself or how it is referenced in your project.
> 
> Source map files are not essential for the proper functioning of Bootstrap or your website. They are used for debugging purposes, allowing you to map minified or transpiled code back to the original source code during development.

but just for the sake of cleanliness, is there anything I can do about it?

**Comment by cscheid** [comment]
(To be clear, those are not errors, they're warnings. The hint is that they should be showing up in yellow, and not in red)

We recently changed this behavior, and these warnings should no longer be there. What version of quarto are you running?

**Reply by cscheid** [reply, marked as answer]
Specifically, I believe we fixed that in this commit: https://github.com/quarto-dev/quarto-cli/commit/e66d909b281a6679644772027bae0f9aa58ac669

**Reply by richardsprague** [reply]
I'm using 1.3.361.  Not sure I want to upgrade to 1.4 just yet (and flood you with more reports :-) )

**Reply by cscheid** [reply]
> Not sure I want to upgrade to 1.4 just yet (and flood you with more reports :-) )

We 100% don't mind. Frankly, the issue is that you will (and should!) send us messages either way. But on 1.3, they'll more likely be stale like this, which is extra work on everyone's part.

(If you want to minimize the messages, then a readthrough of news/changelog-1.4.md can be especially helpful!)

**Reply by zoushucai** [reply]
@cscheid

**Reply by mcanouil** [reply]
@zoushucai please avoid tagging/pinging people without any messages, simply because you did not get a reply over the weekend.
Thank you!

**Comment by zoushucai** [comment]
Hello, I've encountered a situation where I'm getting similar results. Could you assist me in finding a solution? refer to 
[https://github.com/zoushucai/test01](https://github.com/zoushucai/test01)  and [https://github.com/quarto-dev/quarto-cli/discussions/6778](https://github.com/quarto-dev/quarto-cli/discussions/6778)




I've noticed that the issue may be related to annotations in JS or CSS. It appears that these annotations need to be removed. However, it's worth mentioning that these annotations appear to be generated by Quarto's built-in JS or CSS functionalities.







<img width="1018" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/15843338/7bf30f54-c136-4bfd-81bc-3faa813d8d8c">

**Reply by cscheid** [reply]
This has been fixed in recent v1.4 prereleases.

**Reply by zoushucai** [reply]
Thank you for your reply, but I still have the problem

<img width="1010" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/15843338/26ff1b31-c1d7-4336-9f72-2011d40a2d0c">

This is my environment (mac  m1, Ventura 13.5.1 (22G90))

````
⚡ quarto check                 
Quarto 1.4.358
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.7: OK
      Dart Sass version 1.55.0: OK
      Deno version 1.33.4: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.358
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2023.08
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/zsc/Library/TinyTeX/bin/universal-darwin
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.9.6
      Path: /Library/Developer/CommandLineTools/usr/bin/python3
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

(|) Checking R installation...........载入需要的程辑包：myfun
(\) Checking R installation...........载入需要的程辑包：magrittr
[✓] Checking R installation...........OK
      Version: 4.3.0
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library
      knitr: 1.43
      rmarkdown: 2.24

[✓] Checking Knitr engine render......OK

````

**Reply by cscheid** [reply]
Thanks - I can repro it locally. We attempted a fix here https://github.com/quarto-dev/quarto-cli/commit/e66d909b281a6679644772027bae0f9aa58ac669 but clearly that isn't working.

(With that said, this is just a warning about a source map, it has no effect on the output.)

**Reply by zoushucai** [reply]
Yes, what you said is correct, it does not affect the output I think regular expressions are not comprehensive enough

It could be like this

```
 /*# sourceMappingURL=.....*/
 or
 //# sourceMappingURL=.....

```
Therefore, the [e66d909](https://github.com/quarto-dev/quarto-cli/commit/e66d909b281a6679644772027bae0f9aa58ac669)  repair was not comprehensive enough

```ts
const kSourceMappingRegex = /\/\*\# sourceMappingURL\s*=.* \*\//gi;
const kSourceMappingRegex2 = /\/\/\# sourceMappingURL\s*=.+$/gi;

```

**Reply by cscheid** [reply]
Thanks for spotting that. It'll be fixed soon.

**Comment by zoushucai** [comment]
For example, if there is a comment at the end of a CSS file, the above warning will appear:
```css
/*# sourceMappingURL=debc5d5d77c3f9108843748ff7464032.css.map */
```
I have provided a simple solution in the above project, which is to modify annotations in CSS or JS after rendering by `post-render: compress_jscss.R`. However, I hope to receive your support. Thank you


````yaml


project:
  type: website
  #post-render: compress_jscss.R

website:
  title: "test"
  navbar:
    left:
      - href: index.qmd
        text: Home
      - about.qmd

format:
  html:
    theme: cosmo
    css: styles.css
    toc: true

editor: visual

````
